store/multiversion/store.go

```diff
@@ -212,9 +219,109 @@ func (s *Store) GetReadset(index int) ReadSet {
 	return s.txReadSets[index]
 }
 
-func (s *Store) ValidateTransactionState(index int) []int {
+func (s *Store) SetIterateset(index int, iterateset Iterateset) {
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+
+	s.txIterateSets[index] = iterateset
+}
+
+func (s *Store) GetIterateset(index int) Iterateset {
+	s.mtx.RLock()
+	defer s.mtx.RUnlock()
+
+	return s.txIterateSets[index]
+}
+
+// CollectIteratorItems implements MultiVersionStore. It will return a memDB containing all of the keys present in the multiversion store within the iteration range prior to (exclusive of) the index.
+func (s *Store) CollectIteratorItems(index int) *db.MemDB {
+	sortedItems := db.NewMemDB()
+
+	// get all writeset keys prior to index
+	keys := s.GetAllWritesetKeys()
+	for i := 0; i < index; i++ {
```

Comment:
should we also check `i < len(keys)`?

Reply:
why is that? `i` is used to access the writeset corresponding to a specific transaction index in the multiversion store, theres no guarantee that all indices i < len(keys) would be present this moment, in which case we should skip to the next index. I'll add in an `ok` check for the map value before iterating over the indexedWriteset for explicit checking of presence, but it should still no-op given that a range over a nil slice would no-op, right?